Docs: Relabel and reorganize hosting features as reference (Diátaxis) (#9952)
readthedocs/readthedocs.org: docs/user/hosting.rst

.. old label
.. _Documentation Hosting Features:

> Hosting features provided by Read the Docs

Reviewer: Having RTD in the title is a bit odd? It's already autoappended to each title.

Seems like the original title was better and more explicit, and better for SEO (includes Documentation hosting).

Author: There are similar reasons for why I called it "Special hosting features" in the navigation label.

Alternatively, putting just "Hosting features" made it sound too generic for me. But okay there's a bit of inflation in the language of both suggestions.

We can also simply call it "Documentation hosting features" again, it's not too bad :)

Author: Reverted to old title for now.

==========================================

This article provides a quick overview and reference to how documentation is *hosted* on Read the Docs.
Hosting is optimized with documentation in mind and a number of special features are provided,

> some of which may be customized.

Reviewer: This is super passive voice.. Features are provided? Some can be customized? This is pretty bland language, and can definitely be improved.

Author: Wrote it with less bland language... hopefully maybe eventually :D

.. seealso::

>    :doc:`/reference/features`
>      For a full list of Read the Docs features, see the features reference page.

Reviewer: ```suggestion
   :doc:`Feature reference </reference/features>`
    A complete list of our features.
```

This feels more explicit and clearer? Probably should just rename this page to make it more obvious?

Author: I'll give it a quick rename :+1: The final destination can come later. I'm sure it's better to rename the actual page though than to give explicit labels in the `seealso::` boxes.

Overview of hosting features
----------------------------

Subdomain support
  Every project has a subdomain that is available to serve its documentation based on it's :term:`slug`.
  If you go to ``<slug>.readthedocs.io``, it should show you the latest version of your documentation,
  for example https://docs.readthedocs.io.
  For :doc:`/commercial/index` the subdomain looks like ``<slug>.readthedocs-hosted.com``.

Custom domain support

>   Documentation projects may also define their own domains.

Reviewer: ```suggestion
  Documentation projects can also use their own domains.
```

  A project may define multiple domains,

>   as well as define which is the *canonical domain*.

Reviewer: This seems oddly specific without explaining what is does? 

```suggestion
  as well as define the *canonical domain* where all other domains will redirect.
```

Author: The "Read more" link below was intended to address the lack of explanation.

  ⏩️ :doc:`Read more </custom-domains>`

Multiple documentation versions

>   The built-in URL structure of Read the Docs is pre-defined to allow for multiple versions and translations to be hosted.

Reviewer: ```suggestion
   We allow for multiple versions and translations to be hosted,
   integrated nicely into the URL of your documentation.
```

>   You can also switch off this mechanism with :doc:`Single-Version projects </single_version>`.

Reviewer: We should never say `mechanism` -- but more importantly, we should explain the benefit rather than saying you can disable it..

```suggestion
  If you only have 1 version and translation,
  we support :doc:`single version projects </single_version>`.
```

>   ⏩️ :doc:`Read more </automatic-redirects>`

Reviewer: This is the wrong link?

Author: It's in anticipation of https://github.com/readthedocs/readthedocs.org/pull/9896 where this page has changed: https://docs--9896.org.readthedocs.build/en/9896/automatic-redirects.html

But it's better with https://docs--9896.org.readthedocs.build/en/9896/versions.html so I've changed it :+1:

Redirect support
  Projects may define their own custom URL redirects rules that trigger in the HTTP layer.

  ⏩️ :doc:`Read more </user-defined-redirects>`

Content Delivery Network (CDN)
  Documentation projects are by nature static HTML pages and assets.
  Hence contents are delivered through a speedy distributed cloud proxy.

  ⬇️ `Read more <Content Delivery Network (CDN) and caching>`__

Sitemaps
  Sitemaps are generated and hosted automatically,
  improving search engine crawling.

  ⬇️ `Read more <Sitemaps>`__

Custom 404s
  A 404 page is provided by default. It may be replaced with a projects own 404 page.

  ⬇️ `Read more <Custom Not Found (404) pages>`__

Automatic and custom robots.txt
  `robots.txt`_ files allow you to customize how your documentation is indexed in search engines.
  We automatically generate one for you, but you can also customize it.

>   ⬇️ `Read more <Custom robots.txt>`__

Reviewer: Does this work?

Author: Nope, using `:ref:<hosting:...>` instead :+1:
